tests/core/data/test_excel_data_node.py

```diff
@@ -263,6 +263,44 @@ def test_write(self, excel_file, default_data_frame, content, columns):
         excel_dn.write(None)
         assert len(excel_dn.read()) == 0
 
+    @pytest.mark.parametrize(
+        "content,columns,sheet_name",
+        [
+            ([{"a": 11, "b": 22, "c": 33}, {"a": 44, "b": 55, "c": 66}], None, "sheet_name"),
+            ([[11, 22, 33], [44, 55, 66]], ["e", "f", "g"], "sheet_name"),
+        ],
+    )
+    def test_write_with_sheet_name(self, excel_file_with_sheet_name, default_data_frame, content, columns, sheet_name):
```

Comment:
This test looks messy to me. 
In some cases, we are testing the `write` function; in some other, we are testing the `write_with_column_names`. It should be split into multiple tests, and the name of the test should be clear enough to understand exactly what is tested.

Reply:
I can only say I have tried my best in the most recent commit 😃

Reply:
🤣
My comment is not specific to your PR or even this test. And that's not a blocking point.
We all (in the back-end team) tend to write too-long unit tests testing multiple things. Sometimes, they are not unitary.

I should create a dedicated issue to spend a couple of days reviewing all the unit tests and trying to improve the overall readability by shortening them, splitting some of them, eventually finding better names, and/or better organizing them in multiple modules and classes...

Let's forget about this comment if that's ok for you. It should be on another issue.

Reply:
too late, already made changes to that 🤣

Reply:
Haha, that s useful anyway. Thx.